harness/determined/deploy/gke/cli.py

```diff
@@ -0,0 +1,596 @@
+                "effect": "NoSchedule",
+            }
+        ]
+        gpu_spec["nodeSelector"] = {"accelerator_type": "gpu"}
+        helm_values["taskContainerDefaults"]["gpuPodSpec"] = gpu_pod_spec
+        cpu_pod_spec = make_spec(helm_values["taskContainerDefaults"], "cpuPodSpec")
+        cpu_spec = cast(Dict, cpu_pod_spec["spec"])
+        cpu_spec["tolerations"] = [
+            {
+                "key": "gpuAvailable",
+                "operator": "Equal",
+                "value": "False",
+                "effect": "NoSchedule",
+            }
+        ]
+        cpu_spec["nodeSelector"] = {"accelerator_type": "cpu"}
+        if args.cpu_coscheduler:
+            cpu_spec["schedulerName"] = "coscheduler"
+        helm_values["taskContainerDefaults"]["cpuPodSpec"] = cpu_pod_spec
+    with (helm_dir / "values.yaml").open("w") as f:
+        yaml.round_trip_dump(helm_values, f)
+
+
+def handle_up(args: argparse.Namespace) -> None:
+    try:
+        validate_args(args)
+    except ValueError as e:
+        exc_str = "Argument Error: {}".format(e)
+        cprint(exc_str, "red")
+        cprint("Failed to create gke cluster", "red")
+        sys.exit(1)
+    if args.gpu_coscheduler or args.preemption:
+        cprint(
+            (
+                "Autoscaling is not supported with the lightweight coscheduling plugin or with the"
+                + " preemptive priority-based scheduler, and so a GPU node pool with {} nodes will"
+                + " be statically allocated. This can be changed by using the default Kubernetes"
+                + "  scheduler, or specifying a different value for --max-gpu-nodes."
+            ).format(args.max_gpu_nodes),
+            "yellow",
+        )
+    if args.cpu_coscheduler:
+        cprint(
+            (
+                "Autoscaling is not supported with the lightweight coscheduling plugin and so a CPU"
+                + " node pool with {} nodes will be statically allocated. This can be changed by"
+                + "  using the default Kubernetes scheduler, or specifying a different value for"
+                + " --max-cpu-nodes."
+            ).format(args.max_cpu_nodes),
+            "yellow",
+        )
+    create_cluster(args)
+    configure_helm(args)
+    if args.cpu_coscheduler:
+        cprint(
+            (
+                "The Priority-Based Gang Scheduler has been enabled for the CPU Agent Pool. To use"
+                + " this feature, please specify the following in your experiment config for"
+                + " commands or CPU Experiments:"
+            ),
+            "yellow",
+        )
+        exp_config = {
+            "environment": {
+                "podspec": {
+                    "metadata": {
+                        "labels": {
+                            "pod-group.scheduling.sigs.k8s.io/name": "<unique task name>",
+                            "pod-group.scheduling.sigs.k8s.io/min-available": "<# of Nodes required>",  # noqa: E501
```

Comment:
This doesn't seem right that when referencing a CPU agent pool we're talking about a number of nodes required, and we're not showing how to set the priority of a task when telling people how to use the priority scheduler. Am I missing something?

Reply:
That really predates me. And honestly I haven't ever heard about CPU coscheduling setups working. I could cut this entire section if it's not useful. 
I'd say the original point of this tool was to avoid writing 8 command lines necessary to make a cluster. Not to try to make it produce all possible working configurations for everyone easily. I'd rather remove the guardrails to allow configurability.

Reply:
@eecsliu probably has the best context on this. Perhaps you could enlighten us on this issue and the ones I've mentioned above so we don't have to relearn anything the hard way? :smile:

Reply:
yeah CPU training with coscheduling is broken because it requires a number of nodes, which you currently have to manually calculate. If you don't set this `pod-group.scheduling.sigs.k8s.io/min-available` field, the coscheduler will ignore it. However, it then becomes no better than the default scheduler.